backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interest/InterestComputationCommand.java

```diff
@@ -415,45 +435,46 @@ private InterestRunId createInterestRunRecord(@NonNull final InterestBonusComput
 	private void computeAndDistributeBonus(@NonNull final InterestComputationRequest request)
 	{
 		Check.assume(request.getComputingMethodType() == ComputingMethodType.SubtractValueOnInterim,
-				"Bonus can only be distributed for" + ComputingMethodType.SubtractValueOnInterim);
+					 "Bonus can only be distributed for" + ComputingMethodType.SubtractValueOnInterim);
 		createInterestRecords(request);
 	}
 
-	private void splitOpenAmount(@NonNull final InvoiceAllocations invoiceAllocations)
+	// TODO: Do we still need this?
+	private void splitOpenAmount(@NonNull final InterimContractAllocations interimContractAllocations)
 	{
-		final ModularContractLogEntry invoice = invoiceAllocations.getInvoiceEntry();
```

Comment:
I don;t think this is needed anymore.
Or do we support several interim contracts?
That's even possible?

Reply:
Multiple interim contracts may be a cause of concern, since we create them for the full amount (qty and price). This opens the door to:
- How do we match shipping notifications (amount wise) on separate interim contracts ? Sequentially, in force contract at the time of shipping, first contract always, etc.
- What do we consider a partial match (this requires interim contract log splitting, so as to have the unmatched amt available for second final invoice).

Based on all these constraints, I'd say we either not allow multiple interim contracts, or always consider the first one (maybe log subsequent interim contracts as informative).